pkg/runner/runner.go

```diff
@@ -119,7 +133,21 @@ func AssembleProgram(cairoProgram *starknet.StarknetProgram, userArgs []starknet
 		return Program{}, nil, nil, fmt.Errorf("cannot get hints: %w", err)
 	}
 
-	entryCodeInstructions, entryCodeHints, err := GetEntryCodeInstructions(mainFunc)
+	entryCodeInstructions, entryCodeHints, currentCodeOffset, builtins, gotGasBuiltin, gotSegmentArena, err := GetEntryCodeInstructions(mainFunc, proofmode)
+
+	program.Builtins = builtins
+	program.GotGasBuiltin = gotGasBuiltin
+	program.GotSegmentArenaBuiltin = gotSegmentArena
+
+	if err != nil {
+		return Program{}, nil, nil, fmt.Errorf("cannot get entry code instructions: %w", err)
+	}
```

Comment:
You should do the check before setting `program` fields

Reply:
got it